pkg/process/statsd/statsd.go

```diff
@@ -6,42 +6,16 @@
 package statsd
 
 import (
-	"net"
-	"os"
-	"strconv"
-
 	"github.com/DataDog/datadog-go/v5/statsd"
 )
 
 // Client is a global Statsd client. When a client is configured via Configure,
 // that becomes the new global Statsd client in the package.
-var Client *statsd.Client
+var Client statsd.ClientInterface = &statsd.NoOpClient{}
```

Comment:
This will allocate a client for anyone who imports this package. Are you sure you want this?

Reply:
Were you able to figure out why this change is necessary?

Reply:
`actual, err := processCheck.run(0, false)` in `process_test.go` calls `ProcessCheck::run` which does
```
statsd.Client.Gauge("datadog.process.containers.host_count", float64(totalContainers), []string{}, 1) //nolint:errcheck
	statsd.Client.Gauge("datadog.process.processes.host_count", float64(totalProcs), []string{}, 1)       //nolint:errcheck
```

without initialising  the statsd client before.

I checked an the size of `NoOpClient{}` is literally 0 (a bit more for the pointer to it)

Reply:
So how was that test passing before?

Reply:
Ok I found the reason, before this calls to `Gauge` fail with `noClientErr("statsd client is nil")` but we do not check the results.
But since this PR switches it to a `ClientInterface` this doesn't `work` anymore and we need to have an (empty) `NoopClient` to have the same behavior as before

Reply:
Great. Thanks for looking into it.